### PR TITLE
2023SDK samples

### PR DESCRIPTION
**Pull request by jmcouffin**
Hi Jeremy, 
I added the 2023 SDK samples. 
Bear with me as it is my first try here.
I guessed it was a simple cut and paste from the SDK, so I acted accordingly.

### CODE REVIEW

**Comment by jeremytammik** [comment]
brilliant! thank you very much. how exactly did you proceed, please? did you check whether there are any old samples that have become obsolete and need to be removed? or simply copied the new ones over the existing?

**Comment by jmcouffin** [comment]
Simply copied over.
I did not do a diff analysis

**Comment by jeremytammik** [comment]
okey-doke... i'll check it out a bit, then, first...

File: SDK/Samples/AnalyticalSupportData_Info/CS/AnalyticalSupportData_Info.cs [hunk 1]
```diff
@@ -95,80 +96,94 @@ public DataTable ElementInformation
...
+      /// a data table which store all the required information
+      /// </returns>
+      private DataTable StoreInformationInDataTable(ElementSet selectedElements)
+      {
+         DataTable informationTable = CreatDataTable();
+
+         foreach (Element element in selectedElements)
+         {
+            // Get  
+            AnalyticalElement analyticalModel = null;
```
Comment (jmcouffin) [inline SDK/Samples/AnalyticalSupportData_Info/CS/AnalyticalSupportData_Info.cs:115]: noticable change

File: SDK/Samples/BoundaryConditions/CS/BoundaryConditionsData.cs [hunk 2]
```diff
@@ -195,84 +195,101 @@ private void SetBCHostMap(Autodesk.Revit.DB.Element element)
...
-        /// All the parameter default as Fixed.
-        /// </summary>
-        /// <param name="hostElement"> 
-        /// structural element which provide the analytical line end reference
-        /// </param>
-        /// <returns> the created Point BoundaryConditions Element</returns>
-        private Autodesk.Revit.DB.Structure.BoundaryConditions CreatePointBC(Autodesk.Revit.DB.Element hostElement)
-        {
-            if (!(hostElement is FamilyInstance))
+      AnalyticalElement GetAnalyticalElement(Autodesk.Revit.DB.Element element)
```
Comment (jmcouffin) [inline SDK/Samples/BoundaryConditions/CS/BoundaryConditionsData.cs:206]: noticable change

File: SDK/Samples/BoundaryConditions/CS/BoundaryConditionsData.cs [hunk 3]
```diff
@@ -195,84 +195,101 @@ private void SetBCHostMap(Autodesk.Revit.DB.Element element)
...
+      /// </param>
+      /// <returns> the created Point BoundaryConditions Element</returns>
+      private Autodesk.Revit.DB.Structure.BoundaryConditions CreatePointBC(Autodesk.Revit.DB.Element hostElement)
+      {
+         if (!(hostElement is FamilyInstance))
+         {
+            return null;
+         }
+         FamilyInstance familyInstance = hostElement as FamilyInstance;
+         AnalyticalElement analyticalModel = GetAnalyticalElement(hostElement);
```
Comment (jmcouffin) [inline SDK/Samples/BoundaryConditions/CS/BoundaryConditionsData.cs:241]: noticable change

**Comment by jeremytammik** [comment]
Thank you for checking. So far, I just compared the directories and determined the following additions stored in my local file `diff_2022_2023.txt`:

```
377a378,380
> Samples/SheetToView3D/CS/Properties
> Samples/SheetToView3D/CS
> Samples/SheetToView3D
584a588,590
> Samples/Events/SelectionChanged/CS/Properties
> Samples/Events/SelectionChanged/CS
> Samples/Events/SelectionChanged
646a653,655
> Samples/ContextualAnalyticalModel/CS/Properties
> Samples/ContextualAnalyticalModel/CS
> Samples/ContextualAnalyticalModel
833c842
< 2022
---
> 2023
```

No removals.

**Comment by jeremytammik** [comment]
So, maybe we have three new samples added to the Revit 2023 SDK? 

- SheetToView3D
- SelectionChanged
- ContextualAnalyticalModel

**Comment by jmcouffin** [comment]
> So, maybe we have three new samples added to the Revit 2023 SDK?
> 
> * SheetToView3D
> * SelectionChanged
> * ContextualAnalyticalModel

Yes, I looked through 1/4 quickly. I noticed most of the changes went from Id.Int to ElementId and some analytical model new ways of grabbing stuff

**Comment by jeremytammik** [comment]
what do you mean by `1/4`? :-)

**Comment by jmcouffin** [comment]
I went through one quarter of the cs and vb file looking at the different for anything out of the ordinary.

**Comment by jeremytammik** [comment]
great! thank you!

**Comment by jeremytammik** [comment]
i completed my comparison now as well. here is my diff file `diff_2022_2023.txt`, now listing both directory differences and individual file differences: 

```
Comparing directories only:

% diff du2022.txt du2023.txt

--
377a378,380
> Samples/SheetToView3D/CS/Properties
> Samples/SheetToView3D/CS
> Samples/SheetToView3D
584a588,590
> Samples/Events/SelectionChanged/CS/Properties
> Samples/Events/SelectionChanged/CS
> Samples/Events/SelectionChanged
646a653,655
> Samples/ContextualAnalyticalModel/CS/Properties
> Samples/ContextualAnalyticalModel/CS
> Samples/ContextualAnalyticalModel
833c842
< 2022
--

Comparing files and directories:

find . | sort  > ff2022.txt
find . | sort  > ff2023.txt
diff ff2022.txt ff2023.txt >> diff_2022_2023.txt

--
> 2023
780d779
< ./Samples/AnalysisVisualizationFramework/DistanceToSurfaces/CS/Properties
787d785
< ./Samples/AnalysisVisualizationFramework/MultithreadedCalculation/CS/Properties
792d789
< ./Samples/AnalysisVisualizationFramework/SpatialFieldGradient/CS/Properties
834d830
< ./Samples/ArchSample/VB.NET/My Project
942d937
< ./Samples/BrowseBindings/VB.NET/My Project
1007a1003,1025
> ./Samples/ContextualAnalyticalModel
> ./Samples/ContextualAnalyticalModel/CS
> ./Samples/ContextualAnalyticalModel/CS/AddAssociation.cs
> ./Samples/ContextualAnalyticalModel/CS/AnalyticalNodeConnStatus.cs
> ./Samples/ContextualAnalyticalModel/CS/ContextualAnalyticalModel.addin
> ./Samples/ContextualAnalyticalModel/CS/ContextualAnalyticalModel.csproj
> ./Samples/ContextualAnalyticalModel/CS/CreateAnalytcalPanel.cs
> ./Samples/ContextualAnalyticalModel/CS/CreateAnalyticalMember.cs
> ./Samples/ContextualAnalyticalModel/CS/FlipAnalyticalMember.cs
> ./Samples/ContextualAnalyticalModel/CS/MemberForcesAnalyticalMember.cs
> ./Samples/ContextualAnalyticalModel/CS/ModifyPanelContour.cs
> ./Samples/ContextualAnalyticalModel/CS/MoveAnalyticalMemberUsingElementTransformUtils.cs
> ./Samples/ContextualAnalyticalModel/CS/MoveAnalyticalMemberUsingSetCurve.cs
> ./Samples/ContextualAnalyticalModel/CS/MoveAnalyticalNodeUsingElementTransformUtils.cs
> ./Samples/ContextualAnalyticalModel/CS/MoveAnalyticalPanelUsingElementTransformUtils.cs
> ./Samples/ContextualAnalyticalModel/CS/MoveAnalyticalPanelUsingSketchEditScope.cs
> ./Samples/ContextualAnalyticalModel/CS/Properties
> ./Samples/ContextualAnalyticalModel/CS/Properties/AssemblyInfo.cs
> ./Samples/ContextualAnalyticalModel/CS/ReadMe_ContextualAnalyticalModel.rtf
> ./Samples/ContextualAnalyticalModel/CS/ReleaseConditionsAnalyticalMember.cs
> ./Samples/ContextualAnalyticalModel/CS/RemoveAssociation.cs
> ./Samples/ContextualAnalyticalModel/CS/SetOuterContourForPanels.cs
> ./Samples/ContextualAnalyticalModel/CS/Utilities.cs
1045d1062
< ./Samples/CreateBeamsColumnsBraces/VB.NET/My Project
1089d1105
< ./Samples/CreateShared/VB.NET/My Project
1284d1299
< ./Samples/DeleteObject/VB.NET/My Project
1294d1308
< ./Samples/DesignOptionReader/VB.NET/My Project
1312d1325
< ./Samples/DirectionCalculation/CS/Properties
1534a1548,1560
> ./Samples/Events/SelectionChanged
> ./Samples/Events/SelectionChanged/CS
> ./Samples/Events/SelectionChanged/CS/Command.cs
> ./Samples/Events/SelectionChanged/CS/InfoWindow.xaml
> ./Samples/Events/SelectionChanged/CS/InfoWindow.xaml.cs
> ./Samples/Events/SelectionChanged/CS/LogManager.cs
> ./Samples/Events/SelectionChanged/CS/Properties
> ./Samples/Events/SelectionChanged/CS/Properties/AssemblyInfo.cs
> ./Samples/Events/SelectionChanged/CS/ReadMe_SelectionChanged.rtf
> ./Samples/Events/SelectionChanged/CS/SelectionChanged.addin
> ./Samples/Events/SelectionChanged/CS/SelectionChanged.cs
> ./Samples/Events/SelectionChanged/CS/SelectionChanged.csproj
> ./Samples/Events/SelectionChanged/CS/SelectionChangedEventArgsExtension.cs
1651a1678
> ./Samples/FabricationPartLayout/CS/FlipPart.cs
1789d1815
< ./Samples/FindReferencesByDirection/FindColumns/CS/Properties
1796d1821
< ./Samples/FindReferencesByDirection/MeasureHeight/CS/Properties
1821d1845
< ./Samples/FireRating/VB.NET/My Project
2041d2064
< ./Samples/HelloRevit/VB.NET/My Project
2292d2314
< ./Samples/MaterialProperties/VB.NET/My Project
2299d2320
< ./Samples/MaterialQuantities/CS/Properties
2657d2677
< ./Samples/PerformanceAdviserControl/CS/Properties
2668d2687
< ./Samples/PhaseSample/VB.NET/My Project
2721d2739
< ./Samples/PointCloudEngine/CS/PointCloudEngine.XML
2727d2744
< ./Samples/PointCloudEngine/CS/Properties
2892d2908
< ./Samples/RevitCommands/VB.NET/My Project
2975d2990
< ./Samples/RotateFramingObjects/VB.NET/My Project
3030d3044
< ./Samples/RvtSamples/CS/RvtSamples.csproj.user
3150a3165,3172
> ./Samples/SheetToView3D
> ./Samples/SheetToView3D/CS
> ./Samples/SheetToView3D/CS/Properties
> ./Samples/SheetToView3D/CS/Properties/AssemblyInfo.cs
> ./Samples/SheetToView3D/CS/ReadMe_SheetToView3D.rtf
> ./Samples/SheetToView3D/CS/SheetToView3D.addin
> ./Samples/SheetToView3D/CS/SheetToView3D.cs
> ./Samples/SheetToView3D/CS/SheetToView3D.csproj
3203d3224
< ./Samples/SlabProperties/VB.NET/My Project
3302d3322
< ./Samples/StructuralLayerFunction/VB.NET/My Project
3325d3344
< ./Samples/TestFloorThickness/VB.NET/My Project
3333d3351
< ./Samples/TestWallThickness/VB.NET/My Project
3358d3375
< ./Samples/TraverseSystem/CS/TraverseSystem.XML
3381d3397
< ./Samples/TypeSelector/VB.NET/My Project
3499d3514
< ./Samples/Viewers/AnalyticalViewer/VB.NET/My Project
3507d3521
< ./Samples/Viewers/ElementViewer/VB.NET/My Project
3513d3526
< ./Samples/Viewers/RevitViewer/VB.NET/My Project
3527d3539
< ./Samples/Viewers/RoomViewer/VB.NET/My Project
3901c3913
< ./ff2022.txt
---
> ./ff2023.txt
--
```

**Comment by jeremytammik** [comment]
I  have now merged the new SDK on my system, so I will close this pull request. thank you very much for submitting it and supporting this effort! have you already compiled the SDK samples, and set up RvtSamples for the new version? that is next on my list...

**Comment by jmcouffin** [comment]
> have you already compiled the SDK samples, and set up RvtSamples for the new version?

not yet (I built two of them, but then... it did not go as planed) / my visual studio setup is a mess right now, I will have to deal with it first.

**Comment by jeremytammik** [comment]
aha. well, i am progressing. i'll attach my blog post draft for you t see what i am doing...

**Comment by jeremytammik** [comment]
[1946_compile_sdk_2023.md](https://github.com/jeremytammik/RevitSdkSamples/files/8441767/1946_compile_sdk_2023.md)

**Comment by jmcouffin** [comment]
> [1946_compile_sdk_2023.md](https://github.com/jeremytammik/RevitSdkSamples/files/8441767/1946_compile_sdk_2023.md)

very nice, I spent an hour fighting the same fight and ended up leaving it as it is as I had "work" to do. I will try to follow your path tomorrow

**Comment by jeremytammik** [comment]
Yes, painful. And, more or less the same pain every year. With interesting variations each time. I documented it all for you here:

https://thebuildingcoder.typepad.com/blog/2022/04/compiling-the-revit-2023-sdk-samples.html

I wish you good luck and a pleasant experience following those steps.

**Pull request closed without merging**